docs/src/literate/helmholtz.jl

```diff
@@ -0,0 +1,169 @@
+
+function doassemble(cellvalues::CellScalarValues{dim}, facevalues::FaceScalarValues{dim},
+                         K::SparseMatrixCSC, dh::DofHandler) where {dim}
+    b = 1.0
+    f = zeros(ndofs(dh))
+    assembler = start_assemble(K, f)
+    
+    n_basefuncs = getnbasefunctions(cellvalues)
+    global_dofs = zeros(Int, ndofs_per_cell(dh))
+
+    fe = zeros(n_basefuncs) # Local force vector
+    Ke = zeros(n_basefuncs, n_basefuncs) # Local stiffness mastrix
+
+    @inbounds for (cellcount, cell) in enumerate(CellIterator(dh))
+        fill!(Ke, 0)
+        fill!(fe, 0)
+        coords = getcoordinates(cell)
+
+        reinit!(cellvalues, cell)
+        # First we derive the non boundary part of the variation problem from the destined solution `u_ana`
+        # ```math
+        # \int \nabla δu \cdot \nabla u d\Omega
+        # + \int δu \cdot u d\Omega
+        # - \int δu \cdot f d\Omega
+        # ```
+        #+
+
+        for q_point in 1:getnquadpoints(cellvalues)
+            dΩ = getdetJdV(cellvalues, q_point)
+            coords_qp = spatial_coordinate(cellvalues, q_point, coords)
+            f_true = LinearAlgebra.tr(hessian(u_ana, coords_qp)) + u_ana(coords_qp)
+            for i in 1:n_basefuncs
+                δu = shape_value(cellvalues, q_point, i)
+                ∇δu = shape_gradient(cellvalues, q_point, i)
+                fe[i] += (δu * f_true) * dΩ
+                for j in 1:n_basefuncs
+                    u = shape_value(cellvalues, q_point, j)
+                    ∇u = shape_gradient(cellvalues, q_point, j)
+                    Ke[i, j] += (∇δu ⋅ ∇u + δu * u) * dΩ
+                end
+            end
+        end
+
+        # Now we manually add the von Neumann boundary terms
+        # ```math
+        # \int δu \cdot (n \cdot \nabla u - g_2) d\Gamma_2
+        # ```
+        #+
+        for face in 1:nfaces(cell)
+            if onboundary(cell, face) && 
+                   ((cellcount, face) ∈ getfaceset(grid, "left") || 
+                    (cellcount, face) ∈ getfaceset(grid, "bottom"))
+                reinit!(facevalues, cell, face)
+                for q_point in 1:getnquadpoints(facevalues)
+                    coords_qp = spatial_coordinate(facevalues, q_point, coords)
+                    n = getnormal(facevalues, q_point)
+                    g = gradient(u_ana, coords_qp) ⋅ n
+                    dΓ = getdetJdV(facevalues, q_point)
+                    for i in 1:n_basefuncs
+                        δu = shape_value(facevalues, q_point, i)
+                        fe[i] += (δu * g) * dΓ
+                    end
+                end
+            end
+        end
+   
+        celldofs!(global_dofs, cell)
+        assemble!(assembler, global_dofs, fe, Ke)
+    end
+    return K, f
+end;
+
+K, f = doassemble(cellvalues, facevalues, K, dh);
+apply!(K, f, dbcs)
+u = Symmetric(K) \ f;
+
+vtkfile = vtk_grid("helmholtz", dh)
+vtk_point_data(vtkfile, dh, u)
+vtk_save(vtkfile)
+
+using Test
```

Comment:
This could be made source only (like I think we do in the other examples) to not have it show up in the generated documentation, but still have it get tested in CI.

Reply:
Done!